### ISSUE TITLE
How to set the serverClientID needed for serverAuthCode (Switching from iOS)?

### ISSUE THREAD

**Issue by JacobWrenn** [flutter/flutter#57945]
In my previous iOS application, I could get a `serverAuthCode` that I could pass to my own server that was then able to redeem tokens using its own client ID and make API requests on behalf of the user.

I used to be able to specify `GIDSignIn.sharedInstance()?.serverClientID = "12345.apps.googleusercontent.com"` to tell the library the client ID of my server. I would like to know how to set this in Flutter.

**Comment by JacobWrenn** [comment]
Note: I am using the latest version of Flutter's official `google_sign_in` plugin.

**Comment by iapicca** [comment]
Hi @JacobWrenn 
This platform is not meant for assistance on personal code. Please see https://flutter.dev/community for resources and asking questions like this,
you may also get some help if you post it on Stack Overflow.
Closing, as this isn't an issue with Flutter itself. If you disagree, please write in the comments and I will reopen it.
Thank you

**Comment by JacobWrenn** [comment]
Hi @iapicca yes, I agree. However Google's official [google sign in plugin](https://pub.dev/packages/google_sign_in) for Flutter says to leave feedback here. I think that this is a missing feature compared to the library on other platforms (unless I have just not found the right docs).

**Comment by iapicca** [comment]
Hi @JacobWrenn 
the goal of this github repository 
is to improve flutter's framework and first party plugins
through bug reports and feature request;

at first sight your issue seems to be a request 
for assistance on your implementation of `google_sign_in`.

If this is not the case please describe the issue with said plugin in details
Thank you

**Comment by JacobWrenn** [comment]
Ok @iapicca, I will rephrase as a feature request.

`google_sign_in` should have support for setting a `serverClientID` (used as the audience parameter in the OAuth login URL). This is a feature in other first-party google sign in libraries.

An example of how this could look alongside existing options would be:
```dart
final GoogleSignIn google = GoogleSignIn(scopes: ["email"], clientId: "abc123", serverClientId: "def456");
google.signIn();
```

**Comment by ditman** [comment]
The serverClientId parameter has been moved to the native configuration of the app (xml/plist) from the programmatic initialization. 

See the updated example app here: https://github.com/flutter/plugins/pull/2116.

Is the programmatic initialization still required?

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by cyanglaz** [comment]
I think this is a valid proposal and we should implement this. @ditman @stuartmorgan

**Comment by ditman** [comment]
@cyanglaz is this because the `serverClientId` is a per-user value (like an ID Token) rather than a per-app config value?

**Comment by jmagman** [comment]
This should have closed when https://github.com/flutter/plugins/pull/5250 merged. Available in google_sign_in 5.4.0.
